### DISCUSSION TITLE
AWS Lambda deployment support

### DISCUSSION THREAD

**Discussion by mariomnts** [keystonejs/keystone#6806] [Feature requests]
Hi:

I am trying to deploy keystonejs-next to a AWS Lambda function using the `createExpressServer` from `@keystone-next/keystone/system` and the library `serverless-http`. I have this handler which works fine locally replying to HTTP calls to /api/graphql simulating a Lambda environment.

```
const { createSystem, initConfig, createExpressServer } = require('@keystone-next/keystone/system');
const serverless = require('serverless-http');
const apiFile = require('./keystone');
const prismaClient = require('./node_modules/.prisma/client').PrismaClient;

const createKeystoneServer = async () => {
  console.log('✨ Starting Keystone');

  const config = initConfig(apiFile);
  const { getKeystone, graphQLSchema } = createSystem(config);

  const keystone = getKeystone(prismaClient);

  console.log('✨ Connecting to the database');
  await keystone.connect();

  console.log('✨ Creating server');
  const server = await createExpressServer(
    config,
    graphQLSchema,
    keystone.createContext,
  );

  return server;
};

const keystoneServer = createKeystoneServer();

// Handler for AWS Lambda
module.exports.handler = async (event, context) => {
    return serverless(await keystoneServer)(event, context);
};
```

The problem I am seeing is when I try to deploy this to Lambda the node_modules is about ~500MB exceeding the maximum package size of 250MB.

I am trying different options like using webpack to create a minified bundle or to manually remove all the dependencies related to `next` or the admin ui but it's difficult to know which dependencies can actually be removed.

Is there an option to just have the headless graphql endpoint removing all the admin ui not needing in that environment? Thanks and happy to help with the development.

**Comment by jlarmstrongiv** [comment]
I haven’t started the keystone portion of my project yet, but my goal will be the same:  running keystone on AWS Lambda.  Unfortunately, those bundle sizes a real pain—they cannot be changed and are an AWS hard limit.

Rather than dealing with crazy bundling methods or having to cut out portions of my project, I just use serverless container images https://www.serverless.com/blog/container-support-for-lambda since the problem never really goes away.  It’s like having an smartphone or computer lacking the necessary storage space and having to constantly delete old apps or videos 

That said, dependencies like `next` should really be marked as peerDependencies or optionalDependencies.  If the admin ui can be removed, that would be cool too.

Thanks for the example with `createKeystoneServer`!  I’ve been looking for something like that

**Reply by mariomnts** [reply]
You're welcome and yes, I would like a way of being able to remove those dependencies. Let see if someone from the main contributors can point us in that direction.

Thanks for the link to the container support in Lambda, I think I am going to try that path and test execution times to see if they are reasonable.
